tests/unit/project_api/test_project_api.py

```diff
@@ -70,23 +74,23 @@ def test_send_request_no_data(self, url_parts, query_params, no_trailing_slash,
 
     # pylint: disable=too-many-arguments
     @ddt.data(
-        (["part1", "part2"], None, [123], False, api_server_address+"/part1/part2/", {'error': True}),
-        (["part1", "part2"], None, 'qwerty', True, api_server_address+"/part1/part2", {'success': True}),
+        (["part1", "part2"], None, [123], False, api_server_address + "/part1/part2/", {'error': True}),
+        (["part1", "part2"], None, 'qwerty', True, api_server_address + "/part1/part2", {'success': True}),
         (
-                ["part1", "part2", "part3"], None, {'data': 11}, False,
-                api_server_address+"/part1/part2/part3/", {'error': True}
+            ["part1", "part2", "part3"], None, {'data': 11}, False,
+            api_server_address + "/part1/part2/part3/", {'error': True}
         ),
         (
-                ["part1"], {'qwe': 'rty'}, {'var1': 1, 'var2': 2}, False,
-                api_server_address+"/part1/?qwe=rty", {'success': True, 'data': [1, 2, 3]}
+            ["part1"], {'qwe': 'rty'}, {'var1': 1, 'var2': 2}, False,
+            api_server_address + "/part1/?qwe=rty", {'success': True, 'data': [1, 2, 3]}
         ),
         (
-                ["part1"], {'qwe': 'rty', 'asd': 'zxc'}, {'stage': 1, 'activity': 2}, False,
-                api_server_address+"/part1/?qwe=rty&asd=zxc", {}
+            ["part1"], url_parameter, {'stage': 1, 'activity': 2}, False,
+            api_server_address + "/part1/?" + urlencode(url_parameter), {}
         ),
         (
-                ["part1"], {'qwe': 'rty', 'asd': 'zxc'}, {'data': None}, True,
-                api_server_address+"/part1?qwe=rty&asd=zxc", {}
+            ["part1"], url_parameter, {'data': None}, True,
+            api_server_address + "/part1?" + urlencode(url_parameter), {}
         ),
```

Comment:
urlencode doesn't give consistant result the keys can come in different order so the test were failing I just introduced it to make it consistent and tried to be DRY